rllib/offline/json_reader.py

```diff
@@ -235,30 +252,76 @@ def _next_file(self) -> FileType:
             path = random.choice(self.files)
         return self._try_open_file(path)
 
+    def _from_json(self, data: str) -> SampleBatchType:
+        if isinstance(data, bytes):  # smart_open S3 doesn't respect "r"
+            data = data.decode("utf-8")
+        json_data = json.loads(data)
```

Comment:
offline training json file may be a few GBs big.
we are loading the whole thing in one shot here right?

Reply:
No, `data` is always only one single line.

Reply:
I see.